velox/vector/arrow/Bridge.cpp

```diff
@@ -474,15 +476,25 @@ void exportNulls(
     out.null_count = 0;
     return;
   }
-  if (!rows.changed()) {
-    holder.setBuffer(0, vec.nulls());
-    out.null_count = vec.getNullCount().value_or(-1);
-    return;
+
+  auto nulls = vec.nulls();
+
+  // If we're only exporting a subset, create a new validity buffer.
+  if (rows.changed()) {
+    nulls = AlignedBuffer::allocate<bool>(out.length, pool);
+    gatherFromBuffer(*BOOLEAN(), *vec.nulls(), rows, *nulls);
+  }
+
+  // Set null counts.
+  if (!rows.changed() && (vec.getNullCount() != std::nullopt)) {
+    out.null_count = *vec.getNullCount();
```

Comment:
Maybe this is wrong, but my impression was always that null count isnt always up to date and is best effort - maybe we should just use countNulls.

Reply:
I suppose I was uncertain if we ever unset it after setting it if some operation takes place on the vector. I will take your word on this though.